Add OpenHands Commercial Open Source Strategy documentation (#268)
All-Hands-AI/docs: overview/strategy.mdx

- **Goal:** Help organizations safely leverage AI at scale to modernize, refactor, and maintain complex systems across 100s of development teams.
- **User personas:**
    - **Developers w/ AI experience:** Developers who already may be familiar with pairing with AI to accomplish work, but ready to begin delegating wholesale tasks to AI agents for maximum productivity.
    - **Agent Builders:** Developers on the edge of AI engineering who are looking to integrate agents into workflows or applications.
- **Buyer personas:** AI / Platform Engineering Leaders
- **Use cases:**
    - "An OpenHands agent for every developer": Multi-user collaboration, scalability, governance, and security
    - Refactoring and code modernization
    - Embedding AI agents into apps and workflows
- **Relevant Projects/Products:**
  - OpenHands Software Agent SDK
  - OpenHands CLI
  - OpenHands Enterprise

## Dividing Lines Between OSS and Commercial

_OpenHands' CEO and founder discusses the lines between OSS and Commercial in this blog post: [Walking the Line with Commercial Open Source (June 2025)](https://openhands.dev/blog/walking-the-line-with-commercial-open-source)_

> OpenHands follows an **Open Core** model:
> 
> 
> - **Open Source Core:**
>     
>     The foundation of OpenHands is open and free for all developers. Most new features default to open source unless they explicitly solve enterprise-specific needs.
>     
> - **Enterprise Offering:**
>     
>     Focused on mid-to-large enterprises, addressing needs around multi-user management, scalability, and security.
>     
> 
> 
> Our investment is roughly **50/50** between open source and commercial development.

Reviewer: ```suggestion

- **Open Source AI Software Engineer:**
    
    The OpenHands agent will always be open and free. Open Source interfaces to it are designed to serve the needs of individual developers.
    
- **Enterprise Offering:**
    
    Focused on mid-to-large enterprises, addressing needs around multi-user management, scalability, enterprise governance and security.
    

Our investment is roughly **50/50** between open source and commercial development. Most new features default to open source unless they explicitly solve enterprise-specific needs.
```

How do you feel about something like this, @rbren ?

I think this tiny nit comes down to the same specifics (same scopes), I just feel it may make a difference in how people hear it. ("Open core" means a lot of things, and it doesn't have a good reputation in OSS communities, for good reasons.)

From the perspective of the Open Source project, a problem we face is to keep trust from potential contributors, that the OpenHands project is open to them and responsive to them. I do believe people need to trust that its focus is on your original vision, on empowering individuals with this amazing technology.

What if we maybe say it like this?

Reviewer: Just a thought... please let me share this kind of funny example: Cline. For context, Cline is saying, in its blogs, on social media, in interviews, the same refrain: that it's Open Source, that it has BYOK, that nothing phones home. That it doesn't make money on inference. That it's fully private.

Again and again.

That it's Open Source and private and uncompromised. Which yeah, it is, but all these things are in OH too.

And people heard it.

It keeps it **crystal-clear** to its primary audience, developers, and... it's doing that, I think, because something here is a matter of trust: trust that you can use it without worrying.

---

I... should admit that I find this a tad annoying: 

OpenHands has always been **better than Cline**. Better agent, better prompts (I read them all), better behavior; and OpenHands is ahead on full-agent behavior. And, OH had all those things, and also supported `codex` models via Responses API, which last I checked, Cline still hadn't done. Cline had BYOK but not with the latest OpenAI models!

Yet, OH, it seems to me, is _not saying clearly_ most of these benefits. It's... unclear here. Why? Why would this be... unclear?

Please see here why is Cline is saying them. I find it actually hilarious that they even started with Open Source/BYOK when asked about its business model 😅 

[Business model and BYOK](https://www.latent.space/i/168490520/clines-business-model-and-bring-your-own-api-key-approach)

<img width="752" height="252" alt="Image" src="https://github.com/user-attachments/assets/063c4043-e39f-443d-9bb1-e3e86e8f1651" />

<img width="757" height="387" alt="Image" src="https://github.com/user-attachments/assets/dea9bf49-b472-45e8-a671-8e9c34c7c731" />

Reviewer: I mean, to my mind, the Cline thing is, I admit, a little annoying, because OpenHands really had all these things and more.

I would suggest that presenting this as "open core" is maybe unnecessary (it suggests _less_), and maybe just _defining the Open Source project / agent / applications for what they are,_ and/or their advantages are, would be easier to understand, more fun and better overall?

The difference between my attempted suggestion and the existing text is, it seems to me, saying "Open Source" vs "open core"; I'd think the underlying scope is actually the ~same

It's the same as the difference I feel between how Cline and OpenHands seem currently _perceived_, which is between
- saying "Open Source, Uncompromised" 
- and saying "OSS but there's a company now"; not to mention again the obvious "OSS but actually wait you can't do x"

Oof, I feel this maybe didn't make much sense...?

    
    Features designed to solve organizational or managerial challenges are commercial (e.g., organizational management, collaboration, scalability).
    

**Example:**

- Commercial feature: Organizational Management — features for managing OpenHands access across hundreds of developers
- Open Source: MCP support, Secrets Support, Planning Agent mode, etc.

## Unique Differentiators

- **Open Source Foundation and Community:** Transparent, extensible, and a community-driven approach to maintaining OpenHands.
- **Model-Agnostic Architecture:** Works with any LLM — the durable value lies in the agent layer, not the model.
- **Cloud Agents:** Run OpenHands in sandboxed containers on remote agent servers -- whether locally or on cloud infrastructure.
- **Focus on Repetitive Tasks:** Automates ongoing engineering work — refactoring, maintenance, dependency upgrades — where AI delivers compounding value.
- **Dual Persona Alignment:** Serves both individual developers ("Build With") and Agent Builders ("Build On") through a unified ecosystem.

## Strategic Roadmap Pillars

1. **Bet on "Cloud Agents" as the best way to work with AI coding agents**
    

>     Prioritize sandboxed runtimes for scalable, persistent, autonomous systems. When run on a remote agent server, it may be referred to as a "Cloud Agent".

Reviewer: A remote agent-server can be in docker locally, or on a local network, or on a remote VM, or in what is called “OpenHands Cloud”.

Referring to remote agent servers as “Cloud Agents” seems confusing to me. 😅

## Example Use Cases — Repetitive Engineering Tasks

OpenHands is used across a wide range of powerful, scalable use-cases in the SDLC process:

- **Maintenance**
    - Example: 30x throughput on CVE resolution
    - Example: Automatic documentation and release notes
- **Modernization**
    - Example: Adding type annotations to an entire Python codebase
    - Example: Refactoring a monolithic Java application to microservices
- **Migration**
    - Example: Upgrading 1000s of jobs from Spark 2 to Spark 3
    - Example: Moving from Redux to Zustand
- **Tech Debt**
    - Example: Detecting and deleting unused code
    - Example: Adding error handling based on production logs
- **Automated Testing, Bug Fixing, Documentation**, etc.

## Target Customers for OSS and Commercial

### Open Source (OSS)

- **Goal:** Empower individual developers with a model-agnostic, open, and extensible coding agent that integrates seamlessly into their workflows.
- **Primary audience:** Individual developers on the bleeding edge of software development; AI hobbyists.

> - **Use cases:** Personal productivity, ad-hoc local development, experimentation, and contributing to open AI infrastructure.

Reviewer: ```suggestion
- **Use cases:** Personal productivity, local development, experimentation, and contributing to open AI infrastructure.
```

    
- **Enterprise Offering:**
    
    Focused on mid-to-large enterprises, addressing needs around multi-user management, scalability, and security.
    

Our investment is roughly **50/50** between open source and commercial development.

### **Feature Selection Principle**

- **Developer problems → Open source by default:**
    
    Features that solve an individual developer's needs belong in the open source core (e.g., productivity, model integration, developer tooling).
    
- **Buyer problems → Commercial features:**
    
    Features designed to solve organizational or managerial challenges are commercial (e.g., organizational management, collaboration, scalability).
    

**Example:**

- Commercial feature: Organizational Management — features for managing OpenHands access across hundreds of developers
- Open Source: MCP support, Secrets Support, Planning Agent mode, etc.

## Strategic Challenges

1. **Large Product Portfolio**
    
    Balancing focus and resources across a broad range of features and use cases.
    
2. **Driving More Community Contributions**
    

>     Encouraging external contributors while maintaining code quality and stewardship.

Reviewer: ```suggestion
    Keeping the Open Source projects healthy, working on a level-playing field, and growing with contributions from academia, software engineering, and any other expertise.
```

I find it interesting that code quality from external contributors could be a perceived challenge. I wonder what made you think so?

Author: Hi Engel, sorry I didn't mean to have it framed that way. I accepted your suggestion.

- **Focus on Repetitive Tasks:** Automates ongoing engineering work — refactoring, maintenance, dependency upgrades — where AI delivers compounding value.
- **Dual Persona Alignment:** Serves both individual developers ("Build With") and enterprise teams ("Build On") through a unified ecosystem.

## Strategic Roadmap Pillars

1. **Bet on Cloud Agents as the best way to work with AI coding agents**
    
    Prioritize cloud-based runtimes for scalable, persistent, autonomous systems.
    
2. **Shifting towards "Build On" to optimize for repetitive, high-leverage work**
    
    Emphasize ongoing tasks like maintenance and refactoring over one-off code generation.
    
3. **Maintain a high performance, model-agnostic approach**
    
    Continue to integrate multiple models and support extensibility as the LLM ecosystem evolves.
    
4. **Bridge both developer needs and enterprise needs**
    
    Build pathways that help individual developers grow into organizational adopters — "Build With" → "Build On."
    
5. **Foster a thriving Open Source community**
    

>     Grow contributor engagement, maintain transparent governance, and accelerate open development.

Reviewer: 🤔 
Transparent governance sounds good. I believe the intention here is to be the way we work, rather than PR-speak.

I wonder if we use the right metrics for contributor engagement. I think maybe a good mark of a healthy Open Source project is, most of the time, long-term contributors: the ability of the project to attract and keep people who develop deep expertise in an area or another, and can help the project be better.

That is not to say that one-off or rare contributions cannot be great, they can. We have good examples here, since benchmarks can be a one-off and can be very useful, or tricky bug fixes that help UX.

With AI PRs though, with AI agents becoming better and better and easier to fire off and end up with a PR, rare/one-off are becoming way too common place. Too easy - and I think maybe it's not difficult to see that some PR authors don't even know the project, don't use it, don't appear interested in it, they can too easily just fire off an agent on it.

I think maybe this is not, even if it may have been in the past, a good metric of project's health. I've been thinking about this a lot, and part of the solution I think is to improve the agent running on issues and PRs so that it gives us a very good first pass(es). Another part of the solution, though, I would suggest, will have to be to maybe understand that AI PRs can take a heavy toll on maintainer time, and we need to find better signals. 

Both for the AI PRs themselves (this is the WIP I mentioned), and for what it means for Open Source, and for the Open Source commitment. 🤔

Author: Very good feedback @enyst. I think we're all trying to find that mix of metrics to show project health. We could eventually add them to https://prdashboard.openhands.dev/

Reviewer: Ah, nice, thank you! That’s the dashboard I just proposed a PR to: 
- https://github.com/OpenHands/community-pr-dashboard/pull/31

    

## Commitment to Open Source

We believe open source works best with clear stewardship and active participation.

- We maintain the **roadmap** and **good first issues** to guide community contributions.
- We **review pull requests diligently** and recognize valuable contributions.
- We maintain **transparent metrics** across:
    - **Community:** Total contributors and open PRs awaiting review.

>     - **Product Quality:** NPS and SaaS adoption as a proxy for user satisfaction.

Reviewer: @openhands what does NPS stand for, in this context? Respond directly in-thread reply

>     - **Product Quality:** NPS and SaaS adoption as a proxy for user satisfaction.

Reviewer: NPS stands for **Net Promoter Score** — a common customer/user satisfaction metric based on how likely someone is to recommend a product. In this section, it’s being used as a proxy for user satisfaction with the product.

_This reply was generated by an AI assistant (OpenHands) on behalf of the user._

- **Primary audience:** Individual developers on the bleeding edge of software development; AI hobbyists.
- **Use cases:** Personal productivity, local development, experimentation, and contributing to open AI infrastructure.

### Commercial

- **Primary audience:** Mid-to-large enterprises and technical teams.
- **Goal:** Help organizations safely leverage AI at scale to modernize, refactor, and maintain complex systems across 100s of development teams.
- **User personas:**
    - **Developers w/ AI experience:** Developers who already may be familiar with pairing with AI to accomplish work, but ready to begin delegating wholesale tasks to AI agents for maximum productivity.
    - **Agent Engineers:** Developers on the edge of AI engineering who are looking to integrate agents into workflows or applications.
- **Buyer personas:** Engineering Leaders
- **Use cases:**
    - "An OpenHands agent for every developer": Multi-user collaboration, scalability, governance, and security
    - Refactoring and code modernization
    - Embedding AI agents into apps and workflows

## Dividing Lines Between OSS and Commercial

> _OpenHands' CEO and founder discusses the lines between OSS and Commercial in this blog post: [Walking the Line with Commercial Open Source (June 2025)](https://openhands.dev/blog/walking-the-line-with-commercial-open-source)_

Author: It timestamps the article, not the time of writing this doc. Ignoring.
